src/ucm/util/ucm_config.c

```diff
@@ -0,0 +1,163 @@
+    fprintf(stream, "%s%s=%zu\n", UCM_ENV_PREFIX, UCM_ALLOC_ALIGN_VAR,
+            ucm_global_config.alloc_alignment);
+
+    ucm_config_print_bool_doc(stream, "Enable memory events", print_flags);
+    fprintf(stream, "%s%s=%s\n", UCM_ENV_PREFIX, UCM_EN_EVENTS_VAR,
+            ucm_config_bool_to_string(ucm_global_config.enable_events));
+
+    ucm_config_print_bool_doc(stream, "Enable installing malloc hooks",
+                              print_flags);
+    fprintf(stream, "%s%s=%s\n", UCM_ENV_PREFIX, UCM_EN_MALLOC_HOOKS_VAR,
+            ucm_config_bool_to_string(ucm_global_config.enable_malloc_hooks));
+
+    ucm_config_print_bool_doc(stream, "Enable modifying the relocation table",
+                              print_flags);
+    fprintf(stream, "%s%s=%s\n", UCM_ENV_PREFIX, UCM_EN_RELOC_HOOKS_VAR,
+            ucm_config_bool_to_string(ucm_global_config.enable_reloc_hooks));
+}
+
+static char *ucm_config_getenv(const char *name)
+{
+    char var_name[64];
+    snprintf(var_name, sizeof(var_name), "%s%s", UCM_ENV_PREFIX, name);
+    return getenv(var_name);
+}
+
+static void ucm_config_set_value_table(const char *name, const char **table,
+                                       int *value)
+{
+    char *str_value;
+    int i;
+
+    str_value = ucm_config_getenv(name);
+    if (str_value == NULL) {
+        return;
+    }
+
+    for (i = 0; table[i] != NULL; ++i) {
+        if (!strcasecmp(table[i], str_value)) {
+            ucm_global_config.log_level = i;
+            return;
+        }
+    }
+}
+
+static void ucm_config_set_value_bool(const char *name, int *value)
+{
+    char *str_value;
+
+    str_value = ucm_config_getenv(name);
+    if (str_value == NULL) {
+        return;
+    }
+
+    if (!strcasecmp(str_value, "1") || !strcasecmp(str_value, "y") || !strcasecmp(str_value, "yes")) {
+        *value = 1;
+    } else if (!strcasecmp(str_value, "0") || !strcasecmp(str_value, "n") || !strcasecmp(str_value, "no")) {
+        *value = 0;
+    }
+}
+
+static void ucm_config_set_value_size(const char *name, size_t *value)
+{
+    char *str_value, *endptr;
+    size_t n;
+
+    str_value = ucm_config_getenv(name);
+    if (str_value == NULL) {
+        return;
+    }
+
+    n = strtoul(str_value, &endptr, 10);
+    if (*endptr == '\0') {
+        *value = n;
+    }
+}
+
+UCS_STATIC_INIT {
+    ucm_config_set_value_table(UCM_LOG_LEVEL_VAR, ucm_log_level_names,
+                               (int*)&ucm_global_config.log_level);
+    ucm_config_set_value_size(UCM_ALLOC_ALIGN_VAR,
+                              &ucm_global_config.alloc_alignment);
+    ucm_config_set_value_bool(UCM_EN_EVENTS_VAR,
+                              &ucm_global_config.enable_events);
+    ucm_config_set_value_bool(UCM_EN_MALLOC_HOOKS_VAR,
+                              &ucm_global_config.enable_malloc_hooks);
+    ucm_config_set_value_bool(UCM_EN_RELOC_HOOKS_VAR,
+                              &ucm_global_config.enable_reloc_hooks);
+}
```

Comment:
i think we need to discuss this separately, maybe in the context of memory registration cache.